### PR TITLE
Recreated the widgets folder

### LINKED ISSUE #30: Dedup widgets
## Description
All common widgets across screens should be de-duplicated (dedup), copied and imported from lib/widgets/

List of widgets to dedup:
- EmailField
- PasswordField
- TextField (Rename NameField to this)

### PR DESCRIPTION
**Pull request by divy-arpit**
### Description

Include a summary of the change and relevant motivation/context.

**Type of Change:** (Delete irrelevant options)
- Code

**Code/Quality Assurance Only** (Delete irrelevant options)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Tested Successfully on local device.

### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [X] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] Any dependent changes have been merged
- [ ] I have added one/more required dependencies
- [X] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [X] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

**Mention the dependencies/packages/plugins that have been added (if any)**

### CODE REVIEW

File: lib/screens/signup/cubit/signup_cubit.dart [hunk 1]
```diff
@@ -4,7 +4,6 @@ import 'package:ecellapp/core/res/strings.dart';
 import 'package:ecellapp/screens/signup/signup_repository.dart';
 import 'package:equatable/equatable.dart';
 import 'package:meta/meta.dart';
-
```
Comment (werainkhatri) [inline lib/screens/signup/cubit/signup_cubit.dart:7]: reason for removing this?

File: lib/screens/signup/signup.dart [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import 'package:ecellapp/screens/signup/cubit/signup_cubit.dart';
-import 'package:ecellapp/screens/signup/widgets/email_field.dart';
-import 'package:ecellapp/screens/signup/widgets/name_field.dart';
-import 'package:ecellapp/screens/signup/widgets/password_field.dart';
+import 'package:ecellapp/widgets/name_field.dart';
```
Comment (werainkhatri) [inline lib/screens/signup/signup.dart:2]: `name_field.dart` is not used anywhere else, so there is no need to move it.

File: lib/widgets/password_field.dart [hunk 3]
```diff
@@ -13,7 +17,7 @@ class _PasswordFieldState extends State<PasswordField> {
   @override
   void initState() {
     super.initState();
-    _passwordVisible = false;
+    _passwordVisible = true;
```
Comment (werainkhatri) [inline lib/widgets/password_field.dart:20]: 🔴 
the issue didn't state the need to change the initial state of password visibility. Don't deviate from the issue requirements. If you have any change suggestion, discuss with me in the issue before making the changes.

**Review by werainkhatri — CHANGES_REQUESTED** [review]
Do the changes and pull the origin/main before pushing the change commits.

File: lib/screens/signup/signup.dart (hunk 2 above)
Comment (divy-arpit) [reply lib/screens/signup/signup.dart:2]: shouldn't there be two separate fields for name, first name and second name?

Comment (werainkhatri) [reply lib/screens/signup/signup.dart:2]: Pull the latest code from main. There are 2 files made by sid in that commit. You don't have to move that.

**Pull request closed without merging**